### PR TITLE
Fix global tool and SDK tests in release pipeline

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Global tool tests needed the DOTNET_ROOT variable to be set.
SDK tests needed the internal nuget feed URL to be inserted.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml [hunk 1]
```diff
@@ -35,6 +37,28 @@ jobs:
...
         path: '$(Pipeline.Workspace)/releasePipeline/metadata'
 
+    - pwsh:  |
+        Import-Module "$(Build.SourcesDirectory)/build.psm1" -Force
+        New-NugetConfigFile -NugetFeedUrl $(PSInternalNugetFeed) -UserName $(PSInternalNugetFeedUserName) -ClearTextPAT $(PSInternalNugetFeedPAT) -FeedName AzDevOpsFeed -Destination '$(Build.SourcesDirectory)/test/hosting'
+
+        Get-ChildItem '$(Build.SourcesDirectory)/test/hosting'
+
+        # The above cmdlet creates a lower-case nuget.config. There also exists a NuGet.config which we needed to replace.
+        # Hence the following workaround
```
Comment (daxian-dbw) [inline tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml:47]: This template will run on Windows, Linux and macOS. How would this be possible on Windows agent?

File: tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml [hunk 2]
```diff
@@ -35,6 +37,28 @@ jobs:
...
+        Move-Item -Path '$(Build.SourcesDirectory)/test/hosting/nuget.config' -Destination '$(Build.SourcesDirectory)/test/hosting/NuGet.Config' -Force -ErrorAction Continue
+        Get-ChildItem '$(Build.SourcesDirectory)/test/hosting'
+
+        if(-not (Test-Path "$(Build.SourcesDirectory)/test/hosting/NuGet.Config"))
+        {
+            throw "NuGet.Config is not created"
+        }
+        else
+        {
+          Get-Content "$(Build.SourcesDirectory)/test/hosting/NuGet.Config" -Raw
```
Comment (daxian-dbw) [inline tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml:58]: please fix the indentation.

Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml:58]: We should be consistent in the file about using 2 or 4 spaces

File: tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml (hunk 1 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml:47]: This wouldn't but the move-item will be a no-op. I can add a if condition if you prefer.

File: tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml (hunk 2 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml:58]: fixed.

File: tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml (hunk 1 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/release-SDKTests.yml:48]: Added a condition

**Comment by adityapatwardhan** [comment]
@daxian-dbw Please re-review

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**